Update RELEASES.md (#13339)
elastic/apm-server: dev_docs/RELEASES.md

## Day after Feature Freeze

* Trigger release workflow manually

>   * For **patch releases**: run the [`run-patch-release`](https://github.com/elastic/apm-server/actions/workflows/run-patch-release.yml) workflow (select `main` branch, specify the patch release version the release is for).  

Reviewer: ```suggestion
  * For **patch releases**: run the [`run-patch-release`](https://github.com/elastic/apm-server/actions/workflows/run-patch-release.yml) workflow (In "Use workflow from", select `main` branch. Then in "The version", specify the patch release version the release is for).
```

    This workflow will: create the release branch; update version across codebase; commit and create PR targeting the release branch.
    Release notes for patch releases must be manually added:
    * Add a new section to the existing release notes file ([Sample PR](https://github.com/elastic/apm-server/pull/12680)).
    * Review the [changelogs/head](https://github.com/elastic/apm-server/tree/main/changelogs/head.asciidoc) file and move relevant changelog entries from `head.asciidoc` to `release_version.asciidoc` if the change is backported to release_version. If changes do not apply to the version being released, keep them in the `head.asciidoc` file.
    * Review the commits in the release to ensure all changes are reflected in the release notes.
    * Add your PR to the documentation release issue ([Sample Issue](https://github.com/elastic/dev/issues/2485)).

>   * For **minor releases**: run the [`run-minor-release`](https://github.com/elastic/apm-server/actions/workflows/run-minor-release.yml) workflow(select `main` branch, specify the minor release version the release is for).  

Reviewer: ```suggestion
  * For **minor releases**: run the [`run-minor-release`](https://github.com/elastic/apm-server/actions/workflows/run-minor-release.yml) workflow (In "Use workflow from", select `main` branch. Then in "The version", specify the minor release version the release is for).  
```

* A new [tag](https://github.com/elastic/apm-server/releases) will automatically be created on GitHub.

> * Bump the version in anticipation of the next release, e.g. [after 8.13.3 release](https://github.com/elastic/apm-server/pull/13066) bump to 8.13.4. **Prepare this PR ahead of time**, on release date the Release Manager will ping youbut only merge after release when pinged by the release manager.

Reviewer: ```suggestion
* Bump the version in anticipation of the next release, e.g. [after 8.13.3 release](https://github.com/elastic/apm-server/pull/13066) bump to 8.13.4. **Prepare this PR ahead of time**, on release date the Release Manager will ping you but only merge after release when pinged by the release manager.
```

Author: > on release date the Release Manager will ping youbut only merge after release when pinged by the release manager.

Now that I'm reading this sentence again, does it even make sense? @carsonip do you think we should rephrase this? I this so. Something like:

```suggestion
* Bump the version in anticipation of the next release, e.g. [after 8.13.3 release](https://github.com/elastic/apm-server/pull/13066) bump to 8.13.4. **Prepare this PR ahead of time** but only merge it once pinged by the Release Manager on release date.
```
